src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/transforms/charts/gauge.ts

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import type {
+  FormBasedLayer,
+  FormBasedPersistedState,
+  GaugeVisualizationState,
+  PersistedIndexPatternLayer,
+  TextBasedLayer,
+} from '@kbn/lens-common';
+import type { DataViewSpec } from '@kbn/data-views-plugin/common';
+import type { SavedObjectReference } from '@kbn/core/types';
+import type { GaugeState, LensApiState } from '../../schema';
+import { fromColorByValueAPIToLensState, fromColorByValueLensStateToAPI } from '../coloring';
+import type { LensAttributes } from '../../types';
+import { DEFAULT_LAYER_ID } from '../../types';
+import type { DeepMutable, DeepPartial } from '../utils';
+import {
+  addLayerColumn,
+  buildDatasetState,
+  buildDatasourceStates,
+  buildReferences,
+  generateApiLayer,
+  generateLayer,
+  getAdhocDataviews,
+  operationFromColumn,
+} from '../utils';
+import {
+  getMetricAccessor,
+  getSharedChartAPIToLensState,
+  getSharedChartLensStateToAPI,
+} from './utils';
+import type { GaugeStateESQL, GaugeStateNoESQL } from '../../schema/charts/gauge';
+import { fromMetricAPItoLensState } from '../columns/metric';
+import type { LensApiAllMetricOperations } from '../../schema/metric_ops';
+import { getValueApiColumn, getValueColumn } from '../columns/esql_column';
+import { isEsqlTableTypeDataset } from '../../utils';
+
+const ACCESSOR = 'metric_formula_accessor';
```

Comment:
we can cleanup this, no need to call it `metric_foruma_accessor` but something like `gauge_accessor`